### ISSUE TITLE
Array of objects converted to string with custom validator

### ISSUE THREAD

**Issue by rj3d** [sebelga/gstore-node#87]
I'm trying to use a custom validator to verify that a field is an array containing one or more objects. However, the custom validator function is getting passed a string as input instead of the original array being passed in. It seems as if the array is having .toString() called on it before it's passed into the validator. However, if I remove the custom validator function then the correct values are saved to the database.

Code, output, package, and node version information below.

Schema and validator:
```
const itemSchema = new Schema({
    name: { type: 'string', required: true },
    arrOfObjs: {
        type: 'array',
        validate: {
            rule: validateArrOfObjs,
        }
    },
    createdOn: { type: 'string', default: gstore.defaultValues.NOW, write: false, read: false }
});

function validateArrOfObjs(obj, validator) {
    console.log("Input type:");
    console.log(typeof obj);
    console.log("Input is array:");
    console.log(Array.isArray(obj));
    console.log("Array:");
    console.log(obj);
    console.log("First item:");
    console.log(obj[0]);
    console.log("First item type:");
    console.log(typeof obj[0]);

    if (!Array.isArray(obj)) {
        return false;
    }

    var allObjs = obj.every((item) => {
        return item !== null && typeof item === 'object';
    });

    return allObjs;
};

const Item = gstore.model('Item', itemSchema);
```

Testing code:
```
var arr = [
  { name: 'foo' },
  { name: 'bar' },
  { name: 'baz' }
];

var item = new Item( {name: 'test123', arrOfObjs: arr} );

console.log("Array:");
console.log(item.arrOfObjs);
console.log("Array first item:");
console.log(item.arrOfObjs[0]);
console.log("Array first item type:");
console.log(typeof item.arrOfObjs[0]);

console.log("Validating!");
const { error } = item.validate();
if (error !== null) throw error;

item.save();
```

Output:
```
Array:
[ { name: 'foo' }, { name: 'bar' }, { name: 'baz' } ]
Array first item:
{ name: 'foo' }
Array first item type:
object
Validating!
Input type:
string
Input is array:
false
Array:
[object Object],[object Object],[object Object]
First item:
[
First item type:
string
```

yarn package.json
```
{
  "name": "gstore-poc",
  "version": "1.0.0",
  "description": "POC for gstore library",
  "main": "index.js",
  "scripts": {
    "api": "babel-node server.js"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "babel-preset-env": "^1.6.1",
    "body-parser": "^1.18.2",
    "express": "^4.16.2",
    "gstore-node": "^2.1.2",
    "mongoose": "^5.0.0-rc1",
    "morgan": "^1.9.0"
  },
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0"
  },
  "babel": {
    "presets": [
      "env"
    ]
  }
}

```

Node version:
```
node --version
v9.3.0
```

**Comment by rj3d** [comment]
Just checked a few different versions of gstore-node. This bug seems to exist from 2.0.0 onward (I tested 2.0,0, 2.1.2, and 2.1.3), but is not present in 1.5.0. In 1.5.0 I get the following output from the same test code:

```
Array:
[ { name: 'foo' }, { name: 'bar' }, { name: 'baz' } ]
Array first item:
{ name: 'foo' }
Array first item type:
object
Validating!
Input type:
object
Input is array:
true
Array:
[ { name: 'foo' }, { name: 'bar' }, { name: 'baz' } ]
First item:
{ name: 'foo' }
First item type:
object
```

**Comment by sebelga** [comment]
Hi,
Thanks for reporting the bug, this is indeed very strange. I will have a look at it this afternoon :+1:

**Comment by sebelga** [comment]
Hi
I just released a new version (2.1.4) that should fix this bug. Thanks for reporting it! This came after a big refactor and I hadn't tested this case 👍 
Let me know if there is still an issue.
Cheers.

**Comment by rj3d** [comment]
Just tested and everything is working as expected! Thanks for the quick fix, it's much appreciated!
